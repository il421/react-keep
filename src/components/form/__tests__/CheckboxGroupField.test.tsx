import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { Form } from "react-final-form";
import {
  CheckboxGroupField,
  CheckboxGroupFieldProps,
} from "../CheckboxGroupField";

let props: CheckboxGroupFieldProps,
  wrapper: ReactWrapper<CheckboxGroupFieldProps, {}>;

beforeEach(() => {
  props = {
    name: "name",
    options: [
      {
        id: "id1",
        value: "1",
      },
      {
        id: "id2",
        value: "2",
      },
    ],
  };
});

test("should render CheckboxGroupField, and contain input element", () => {
  wrapper = mount<CheckboxGroupFieldProps>(
    <Form initialValues={{}} onSubmit={() => {}}>
      {() => <CheckboxGroupField {...props} />}
    </Form>
  );
  expect(wrapper.debug()).toMatchSnapshot();
  expect(wrapper.find("#id1").exists()).toBeTruthy();
  expect(wrapper.find("label[htmlFor='id1']").exists()).toBeTruthy();
  expect(wrapper.find("#id2").exists()).toBeTruthy();
  expect(wrapper.find("label[htmlFor='id2']").exists()).toBeTruthy();
});
