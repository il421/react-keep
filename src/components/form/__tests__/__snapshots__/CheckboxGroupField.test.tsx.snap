// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckboxGroupField, and contain input element 1`] = `
"<ReactFinalForm initialValues={{...}} onSubmit={[Function: onSubmit]}>
  <CheckboxGroupField name=\\"name\\" options={{...}}>
    <Field name=\\"name\\" options={{...}} component={[Function: FieldAdapter]}>
      <FieldAdapter input={{...}} meta={{...}} options={{...}} render={[undefined]}>
        <CheckboxGroup options={{...}} classNames={[undefined]} selectedKeys=\\"\\" onChange={[Function: onChange]}>
          <FlexBox justifyContent=\\"flex-start\\" className={[undefined]}>
            <div style={{...}} className={[undefined]}>
              <div className={[undefined]}>
                <input id=\\"id1\\" value=\\"id1\\" type=\\"checkbox\\" defaultChecked={false} />
                <label htmlFor=\\"id1\\" onClick={[Function: onClick]} />
              </div>
              <span className={[undefined]}>
                1
              </span>
            </div>
          </FlexBox>
          <FlexBox justifyContent=\\"flex-start\\" className={[undefined]}>
            <div style={{...}} className={[undefined]}>
              <div className={[undefined]}>
                <input id=\\"id2\\" value=\\"id2\\" type=\\"checkbox\\" defaultChecked={false} />
                <label htmlFor=\\"id2\\" onClick={[Function: onClick]} />
              </div>
              <span className={[undefined]}>
                2
              </span>
            </div>
          </FlexBox>
        </CheckboxGroup>
      </FieldAdapter>
    </Field>
  </CheckboxGroupField>
</ReactFinalForm>"
`;
